parser/prism/Helpers.cc

```diff
@@ -183,14 +183,14 @@ pm_node_t *PMK::SymbolFromConstant(core::LocOffsets nameLoc, pm_constant_id_t na
 
     pm_symbol_node_t *symbolNode = allocateNode<pm_symbol_node_t>();
     if (!symbolNode) {
+        free(stable);
```

Comment:
In case of errors we need to free the memory we allocated before returning.

Reply:
Can we not wrap these up in a `unique_ptr` that will call `free` for its deleter, so that we don't have to keep worrying about leaking these?

I think something like the following should work:
```c++
typedef std::unique_ptr<T, decltype([](void *p) { free(p); })> UniqueCPtr<T>;

// ... and then

UniqueCPtr<uint9_t *> stable = something;

// ... and then

some_call_that_needs_stable(stable.get());
```